cmd/system/node.go

```diff
@@ -0,0 +1,138 @@
+package system
+
+import (
+	"fmt"
+	"github.com/alexellis/arkade/pkg/archive"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	cp "github.com/otiai10/copy"
+	"github.com/spf13/cobra"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func getLatestVersion(version, channel string) (*string, error) {
+	resp, err := http.Get(fmt.Sprintf("https://nodejs.org/download/%s/%s", channel, version))
+	if err != nil {
+		return nil, err
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	regex := regexp.MustCompile(`(?m)node-(.*)-linux-.*"`)
+	result := regex.FindStringSubmatch(string(body))
+	if len(result) < 2 {
+		return nil, fmt.Errorf("could not find latest version for %s", version)
+	}
+	return &result[1], nil
+}
+
+func MakeInstallNode() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "node",
+		Short: "Install Node.js",
+		Long:  `Install the Node.js JavaScript runtime ✨🐢🚀✨`,
+		Example: `arkade system install node
+  arkade system install node --version v17.9.0`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("version", "v", "latest", "The version for Node.js to install, either a specific version, 'latest' or ' latest-codename'")
+	command.Flags().StringP("path", "p", "/usr/local/lib/nodejs", "Installation path, where a go subfolder will be created")
```

Comment:
The un-tar path for Nodejs on Linux is:

```
/usr/local/
```

Then the subfolders get created under that i.e. `./lib/...` `./bin/...`

```bash
pi@faasd-pi:~ $ wget https://nodejs.org/dist/v16.14.2/node-v16.14.2-linux-armv7l.tar.xz
--2022-04-11 08:44:57--  https://nodejs.org/dist/v16.14.2/node-v16.14.2-linux-armv7l.tar.xz
Resolving nodejs.org (nodejs.org)... 104.20.23.46, 104.20.22.46, 2606:4700:10::6814:172e, ...
Connecting to nodejs.org (nodejs.org)|104.20.23.46|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18275504 (17M) [application/x-xz]
Saving to: ‘node-v16.14.2-linux-armv7l.tar.xz’

node-v16.14.2-linux 100%[===================>]  17.43M  4.87MB/s    in 3.9s    

2022-04-11 08:45:01 (4.45 MB/s) - ‘node-v16.14.2-linux-armv7l.tar.xz’ saved [18275504/18275504]

pi@faasd-pi:~ $ sudo tar -xf node-v16.14.2-linux-armv7l.tar.xz -C /usr/local/ --strip-components=1
pi@faasd-pi:~ $ node --version
v16.14.2
pi@faasd-pi:~ $ 
```

Please match that style. The archive.UntarP function should cover this use-case.

Reply:
The problem I had, and thats the reason I worked around with the Copy function is that 

archive.untar is not honoring the Folders inside the archive and you get something like this:
![image](https://user-images.githubusercontent.com/38325136/162693881-4ca8a6ed-eef7-40e1-b0af-1862d0ad0cfa.png)

The issue with untar_nested is that you get a subfolder
![image](https://user-images.githubusercontent.com/38325136/162693974-4a08f681-fba1-4db3-8e7a-400ecf2af705.png)

I need a way to copy the content of the archive while honoring the structure and keeping the symlinks.

Maybe I am completly wrong here @alexellis. Help is appreciated.